pkg/ccl/backupccl/restore_planning.go

```diff
@@ -1692,9 +1718,21 @@ func doRestorePlan(
 	if err != nil {
 		return err
 	}
-	description, err := restoreJobDescription(p, restoreStmt, from, incFrom, restoreStmt.Options,
+	var fromDescription [][]string
+	if len(from) == 1 {
```

Comment:
Yeah, it's necessary. I've tried to avoid modifying input arguments because I couldn't keep track of what they meant at given points in the code - that means where we _used_ to mutate `from` to include the fully resolved directories, we no longer do. However, behavior is unchanged in the old-style syntax, so passing from there is just fine.